pgmpy/tests/test_models/test_JunctionTree.py

```diff
@@ -38,3 +42,55 @@ def test_add_cyclic_path_raises_error(self):
 
     def tearDown(self):
         del self.graph
+
+
+class TestJunctionTreeCopy(unittest.TestCase):
+    def setUp(self):
+        self.graph = JunctionTree()
+
+    def test_copy(self):
+        self.graph.add_nodes_from([('a', 'b', 'c'), ('a', 'b') ,('a', 'c')])
+        self.graph.add_edges_from([(('a', 'b', 'c'), ('a', 'b')),
+                                   (('a', 'b', 'c'), ('a', 'c'))])
+        graph_copy = self.graph.copy()
+
+        self.graph.remove_edge(('a', 'b', 'c'), ('a', 'c'))
+        self.assertFalse(self.graph.has_edge(('a', 'b', 'c'), ('a', 'c')))
+        self.assertTrue(graph_copy.has_edge(('a', 'b', 'c'), ('a', 'c')))
+
+        self.graph.remove_node(('a', 'c'))
+        self.assertFalse(self.graph.has_node(('a', 'c')))
+        self.assertTrue(graph_copy.has_node(('a', 'c')))
+
+        self.graph.add_node(('c', 'd'))
+        self.assertTrue(self.graph.has_node(('c', 'd')))
+        self.assertFalse(graph_copy.has_node(('c', 'd')))
+
+    def test_copy1(self):
```

Comment:
@Pratik151 Change the name to something more meaningful. Maybe something like: `test_copy_with_factors` ?